tutorials/roostats/MultivariateGaussianTest.py

```diff
@@ -0,0 +1,146 @@
+## \file
+## \ingroup tutorial_roostats
+## \notebook
+## Comparison of MCMC and PLC in a multi-variate gaussian problem
+##
+## This tutorial produces an N-dimensional multivariate Gaussian
+## with a non-trivial covariance matrix.  By default N=4 (called "dim").
+##
+## A subset of these are considered parameters of interest.
+## This problem is tractable analytically.
+##
+## We use this mainly as a test of Markov Chain Monte Carlo
+## and we compare the result to the profile likelihood ratio.
+##
+## We use the proposal helper to create a customized
+## proposal function for this problem.
+##
+## For N=4 and 2 parameters of interest it takes about 10-20 seconds
+## and the acceptance rate is 37%
+##
+## \macro_image
+## \macro_output
+## \macro_code
+##
+## \date July 2022
+## \authors Artem Busorgin, Kevin Belasco and Kyle Cranmer (C++ version)
+
+import ROOT
+
+dim = 4
+nPOI = 2
+
+# let's time this challenging example
+t = ROOT.TStopwatch()
+t.Start()
+
+xVec = []
+muVec = []
+poi = set()
+
+# make the observable and means
+for i in range(dim):
+    name = "x{}".format(i)
+    x = ROOT.RooRealVar(name, name, 0, -3, 3)
+    xVec.append(x)
+
+    mu_name = "mu_x{}".format(i)
+    mu_x = ROOT.RooRealVar(mu_name, mu_name, 0, -2, 2)
+    muVec.append(mu_x)
+
+# put them into the list of parameters of interest
+for i in range(nPOI):
+    poi.add(muVec[i])
+
+# make a covariance matrix that is all 1's
+cov = ROOT.TMatrixDSym(dim)
+for i in range(dim):
+    for j in range(dim):
+        if i == j:
+            cov[i, j] = 3.0
+        else:
+            cov[i, j] = 1.0
+
+# now make the multivariate Gaussian
+mvg = ROOT.RooMultiVarGaussian("mvg", "mvg", xVec, muVec, cov)
```

Comment:
Note to myself: we could implement a nice Pythonization here later, where the `RooMultiVarGaussian` can also take a NumPy array in Python.